grpc/oob/diagnostic/screenshot.go

```diff
@@ -0,0 +1,144 @@
+package diagnostic
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/bmc-toolbox/bmclib/v2"
+	"github.com/bmc-toolbox/bmclib/v2/bmc"
+	"github.com/bmc-toolbox/bmclib/v2/providers"
+	"github.com/go-logr/logr"
+	"github.com/prometheus/client_golang/prometheus"
+	v1 "github.com/tinkerbell/pbnj/api/v1"
+	common "github.com/tinkerbell/pbnj/grpc/oob"
+	"github.com/tinkerbell/pbnj/pkg/metrics"
+	"github.com/tinkerbell/pbnj/pkg/repository"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type Action struct {
+	common.Accessory
+	ScreenshotRequest *v1.ScreenshotRequest
+}
+
+// WithLogger adds a logr to an Action struct.
+func WithLogger(l logr.Logger) Option {
+	return func(a *Action) error {
+		a.Log = l
+		return nil
+	}
+}
+
+// WithScreenshotRequest adds ScreenshotRequest to an Action struct.
+func WithScreenshotRequest(in *v1.ScreenshotRequest) Option {
```

Comment:
This feels like it should be required always, no? if that is the case, you might consider making it explicit as a dependency in the `GetScreenshot` function.